src/Controller/Controller.php

```diff
@@ -292,7 +292,7 @@ public function loadComponent(string $name, array $config = []): Component
      * @param string $name Property name
      * @return \Cake\Controller\Component|\Cake\ORM\Table|null
      */
-    public function __get(string $name)
+    public function __get(string $name): Component|Table|null
```

Comment:
Since return types cannot be widened perhaps we should leave out the types for magic methods?

Reply:
Should we set it to `mixed` to so a return value is required since no type declaration is `mixed|void`?

Reply:
It seems that is the function signature from php docs:
> public __get(string $name): mixed

Reply:
I guess we should add `mixed` then.